src/transaction.ts

```diff
@@ -6,35 +6,53 @@ import {
   TransactionBuilder,
   xdr,
 } from "stellar-base";
+import { SorobanRpc } from "./soroban_rpc";
 
 // TODO: Transaction is immutable, so we need to re-build it here. :(
 export function assembleTransaction(
   raw: Transaction | FeeBumpTransaction,
   networkPassphrase: string,
-  simulated: Array<null | {
-    footprint: Buffer | string | xdr.LedgerFootprint;
-    auth: Array<Buffer | string | xdr.ContractAuth>;
-  }>,
+  simulation: SorobanRpc.SimulateTransactionResponse,
 ): Transaction {
   if ("innerTransaction" in raw) {
     // TODO: Handle feebump transactions
     return assembleTransaction(
       raw.innerTransaction,
       networkPassphrase,
-      simulated,
+      simulation,
     );
   }
 
-  if (simulated.length !== raw.operations.length) {
+  if (
+    raw.operations.length !== 1 ||
+    raw.operations[0].type !== "invokeHostFunction"
+  ) {
     throw new Error(
-      "number of simulated operations not equal to number of transaction operations",
+      "unsupported operation type, must be only one InvokeHostFunctionOp in the transaction.",
+    );
+  }
+
+  const rawInvokeHostFunctionOp: any = raw.operations[0];
+
+  if (
+    !rawInvokeHostFunctionOp.functions ||
+    !simulation.results ||
+    rawInvokeHostFunctionOp.functions.length !== simulation.results.length
+  ) {
+    throw new Error(
+      "preflight simulation results do not contain same count of HostFunctions that InvokeHostFunctionOp in the transaction has.",
```

Comment:
I don't understand the intricacies of the multi-op host functions yet: why is the length enforced to be 1 above, but that isn't enforced here? I guess it's implicit, but I just want to make sure.

Related: is there no way to simulate transactions with multiple `invokeHostOp`erations?

Reply:
@dmkozh , can you confirm with the latest changes for multi-function InvokeHostFunctionOp, the limitation for a single InvokeHostFunctionOp in Tx is still being enforced? I thought have seen that mentioned in chat and was assuming that limitation on validation of tx here.

Reply:
ohhhh yeah sorry, you're right :facepalm: one op per tx, but the op can have many fns. got it! 

:laughing: too many layers :cake: